htmd/adaptive/adaptive.py

```diff
@@ -154,7 +161,25 @@ def run(self):
                 logger.info("Epoch 0, generating first batch")
                 self._init()
                 if not self.dryrun:
-                    self.app.submit(natsorted(glob(path.join(self.inputpath, "e1s*"))))
+                    if self.mps > 0:
+                        from jobqueues.slurmqueue import SlurmQueue
+
+                        if not isinstance(self.app, SlurmQueue):
+                            raise ValueError(
+                                "Multiple processes per simulation only supported with SlurmQueue."
+                            )
+                        input_dirs = natsorted(glob(path.join(self.inputpath, "e1s*")))
+                        for i in range(0, len(input_dirs), self.mps):
+                            if len(input_dirs[i : i + self.mps]) > 1:
```

Comment:
I'm rather sure you will get an index error here if your simulations are not divisible by self.mps
Use `min(i+self.mps, len(input_dirs))` to get the max of the range

Reply:
```py
batch = input_dirs[i:min(i+self.mps, len(input_dirs))]
self.app.submit(batch, nvidia_mps=len(batch) > 1)
```
Then you skip that else-if as well